x-pack/test/api_integration/apis/ml/data_frame_analytics/evaluate.ts

```diff
@@ -0,0 +1,153 @@
+      evaluation: {
+        regression: {
+          actual_field: 'stab',
+          predicted_field: 'ml.stab_prediction',
+          metrics: {
+            r_squared: {},
+            mse: {},
+            msle: {},
+            huber: {},
+          },
+        },
+      },
+    },
+    classification: {
+      index: generateDestinationIndex(`classification_${currentTime}`),
+      evaluation: {
+        classification: {
+          actual_field: 'y',
+          predicted_field: 'ml.y_prediction',
+          metrics: { multiclass_confusion_matrix: {}, accuracy: {}, recall: {} },
+        },
+      },
+    },
+  };
+  const jobAnalysis: any = {
+    classification: {
+      source: {
+        index: ['ft_bank_marketing'],
+        query: {
+          match_all: {},
+        },
+      },
+      analysis: {
+        classification: {
+          dependent_variable: 'y',
+          training_percent: 20,
+        },
+      },
+    },
+    regression: {
+      source: {
+        index: ['ft_egs_regression'],
+        query: {
+          match_all: {},
+        },
+      },
+      analysis: {
+        regression: {
+          dependent_variable: 'stab',
+          training_percent: 20,
+        },
+      },
+    },
+  };
+
+  interface TestConfig {
+    jobType: string;
+    config: DeepPartial<DataFrameAnalyticsConfig>;
+    eval: any;
+  }
+
+  const testJobConfigs: TestConfig[] = ['regression', 'classification'].map((jobType, idx) => {
+    const analyticsId = `${jobType}_${currentTime}`;
+    return {
+      jobType,
+      config: {
+        id: analyticsId,
+        description: `Testing ${jobType} evaluation`,
+        dest: {
+          index: generateDestinationIndex(analyticsId),
+          results_field: 'ml',
+        },
+        analyzed_fields: {
+          includes: [],
+          excludes: [],
+        },
+        model_memory_limit: '60mb',
+        ...jobAnalysis[jobType],
+      },
+      eval: jobEval[jobType],
+    };
+  });
+
+  async function createJobs(mockJobConfigs: TestConfig[]) {
+    for (const jobConfig of mockJobConfigs) {
+      await ml.api.createAndRunDFAJob(jobConfig.config as DataFrameAnalyticsConfig);
+    }
+  }
+
+  describe('POST data_frame/_evaluate', () => {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/bm_classification');
+      await esArchiver.loadIfNeeded('ml/egs_regression');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      await createJobs(testJobConfigs);
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    testJobConfigs.forEach((testConfig) => {
+      describe(`EvaluateDataFrameAnalytics ${testConfig.jobType}`, async () => {
+        it(`should evaluate ${testConfig.jobType} analytics job`, async () => {
+          const { body } = await supertest
+            .post(`/api/ml/data_frame/_evaluate`)
+            .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
```

Comment:
Would be good to have two additional test cases to cover some API privileges:
- the API should work exactly the same for ML_VIEWER
- the API should give a `403` for ML_UNAUTHORIZED

I think we don't necessarily need to repeat those permission test cases for all testJobConfigs but just for one of them.

Reply:
Good catch - added in 42e025353c418cc5ed2b8157f53df0fcff352bfc